File: ts-sdk/src/sdk/models/operations/getcollaborationsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCollaborationsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=collaboration_id" })
  collaborationId: string;
}


export class GetCollaborationsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetCollaborationsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetCollaborationsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCollaborationsIdQueryParams;
}


export class GetCollaborationsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  collaboration?: shared.Collaboration;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getcollaborationsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetCollaborationsIDPathParams struct {
	CollaborationID string `pathParam:"style=simple,explode=false,name=collaboration_id"`
}

type GetCollaborationsIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetCollaborationsIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetCollaborationsIDPathParams
	QueryParams GetCollaborationsIDQueryParams
}

type GetCollaborationsIDResponse struct {
	ClientError   *shared.ClientError
	Collaboration *shared.Collaboration
	ContentType   string
	StatusCode    int64
}
